packages/client/src/components/app/table/Table.svelte

```diff
@@ -42,6 +42,7 @@
   $: {
     rowSelectionStore.actions.updateSelection(
       $component.id,
+      selectedRows.length ? selectedRows[0].tableId : "",
```

Comment:
Setting the table id here, this seems to be fine because the store gets reset when the data provider above it gets changed. Still unsure of why this is the case, but it does.